File: src/Easy/28.Find_the_Index_of_the_First_Occurrence_in_a_String/solution.rb
# @param {String} haystack
# @param {String} needle
# @return {Integer}
def str_str(haystack, needle)
  return 0 if needle.empty?

  index = 0

  while index <= haystack.length - needle.length
    index2 = 0

    while index2 < needle.length &&
          haystack[index + index2] == needle[index2]
      index2 += 1
    end

    return index if index2 == needle.length
    index += 1
  end

  -1
end

puts str_str("mississippi", "issip")

File: src/Easy/28.Find_the_Index_of_the_First_Occurrence_in_a_String/solution.ts
// "mississippi"
// "issip"
function strStr(haystack: string, needle: string): number {
  for (let i = 0; i < haystack.length; i++)  {
    if (haystack[i] === needle[0]) {
      let substr = haystack.slice(i, i + needle.length);
      if (substr === needle) return i;
    }
  }

  return -1;
};